client/tests/phpunit/Pact/SiriusApiGatewayClientTest.php

```diff
@@ -0,0 +1,244 @@
+
+        $builder->verify();
+
+        self::assertStringContainsString(
+            $this->reportPdfUuid,
+            $result->getBody()->getContents()
+        );
+    }
+
+    private function generateReportPdfPactBuilder(string $caseRef)
+    {
+        $matcher = new Matcher();
+
+        $multipartRequest = (new MultipartPactRequest())
+            ->addPart('report_file', 'c29tZV9jb250ZW50')
+            ->addPart('report', [
+                'data' => [
+                    'type' => 'reports',
+                    'attributes' => [
+                        'reporting_period_from' => $matcher->dateISO8601('2018-05-14'),
+                        'reporting_period_to' => $matcher->dateISO8601('2019-05-13'),
+                        'year' => $matcher->regex('2018', '[0-9]{4}'),
+                        'date_submitted' => $matcher->dateTimeISO8601('2019-06-20T00:00:00+01:00'),
+                        'type' => $matcher->regex('PF', 'PF|HW|NDR'),
+                        'submission_id' => $matcher->regex(9876, '\d+')
+                    ]
+                ]
+            ]);
+
+        // Create your expected request from the consumer.
+        $request = (new ConsumerRequest())
+            ->setMethod('POST')
+            ->setPath(sprintf('/clients/%s/reports', $caseRef))
+            ->setHeaders(
+                [
+                    'Content-Type' =>
+                        $matcher->regex(
+                            'multipart/form-data; boundary=5872fc54a8fa5f5be65ee0af590d1ae813a1b091',
+                            'multipart\/form-data.*'
+                        )
+                ]
+            )
+            ->setBody($matcher->regex($multipartRequest->getExampleBody(), $multipartRequest->getRegex('report')));
+
+        // Create your expected response from the provider.
+        $response = new ProviderResponse();
+        $response
+            ->setStatus(201)
+            ->addHeader('Content-Type', 'application/json')
+            ->setBody([
+                'data' => [
+                    'type' => 'reports',
+                    'id' => $matcher->uuid($this->reportPdfUuid),
+                    'attributes' => [
+                        'reporting_period_from' => $matcher->dateISO8601(),
+                        'reporting_period_to' => $matcher->dateISO8601(),
+                        'year' => $matcher->regex('2018', '[0-9]{4}'),
+                        'date_submitted' => $matcher->dateTimeWithMillisISO8601(),
+                        'type' => $matcher->regex('PF', 'PF|HW|NDR'),
+                        'submission_id' => $matcher->integer(9876)
+                    ]
+                ],
+            ]);
+
+        // Create a configuration that reflects the server that was started. You can create a custom MockServerConfigInterface if needed.
+        $config  = new MockServerEnvConfig();
+        $builder = new InteractionBuilder($config);
```

Comment:
Might be worth moving this into the constructor as `$this->builder` to avoid duplication?